tagmanager/src/tm_workspace.c

```diff
@@ -823,514 +728,333 @@ fill_find_tags_array (GPtrArray *dst, const GPtrArray *src,
  @param partial Whether partial match is allowed.
  @param lang Specifies the language(see the table in parsers.h) of the tags to be found,
              -1 for all
- @return Array of matching tags. Do not free() it since it is a static member.
+ @return Array of matching tags.
 */
-const GPtrArray *
-tm_workspace_find_scoped (const char *name, const char *scope, TMTagType type,
-		TMTagAttrType *attrs, gboolean partial, langType lang, gboolean global_search)
+GPtrArray *tm_workspace_find(const char *name, const char *scope, TMTagType type,
+	TMTagAttrType *attrs, gboolean partial, langType lang)
 {
-	static GPtrArray *tags = NULL;
+	GPtrArray *tags;
+	guint tagCount;
 
-	if (tags)
-		g_ptr_array_set_size (tags, 0);
-	else
-		tags = g_ptr_array_new ();
+	tags = g_ptr_array_new();
+
+	fill_find_tags_array(tags, theWorkspace->tags_array, name, NULL, type, partial, lang);
+	fill_find_tags_array(tags, theWorkspace->global_tags, name, NULL, type, partial, lang);
```

Comment:
Ah, yes - it used to be called from find_scope_members() but it got removed in one of the patches so it's really unnecessary now.